In [1]:
import pandas as pd
titanic = pd.read_csv("../data/titanic_propre.csv")
titanic

,survived,pclass,sex,age,sibsp,parch,fare,embark_town,alone
0,0,3,male,22.0,1,0,7.2500,Southampton,False
1,1,1,female,38.0,1,0,71.2833,Cherbourg,False
2,1,3,female,26.0,0,0,7.9250,Southampton,True
3,1,1,female,35.0,1,0,53.1000,Southampton,False
4,0,3,male,35.0,0,0,8.0500,Southampton,True
...,...,...,...,...,...,...,...,...,...
886,0,2,male,27.0,0,0,13.0000,Southampton,True
887,1,1,female,19.0,0,0,30.0000,Southampton,True
888,0,3,female,21.5,1,2,23.4500,Southampton,False
889,1,1,male,26.0,0,0,30.0000,Cherbourg,True


In [2]:
survie_par_classe_et_sexe = (titanic.groupby(["pclass", "sex"])["survived"].mean() * 100).round(1)
survie_par_classe_et_sexe

pclass  sex   
1       female    96.8
        male      36.9
2       female    92.1
        male      15.7
3       female    50.0
        male      13.5
Name: survived, dtype: float64

**Réponse 1 :** Oui, dans les trois classes. En 1re classe, 96.8 % des femmes survivent contre 36.9 % des hommes ; en 2e classe 92.1 % contre 15.7 % ; en 3e classe 50.0 % contre 13.5 %. L'écart est énorme partout.